contracts/Collector.sol

```diff
@@ -68,17 +86,27 @@ contract Collector is ICollector{
     }
 
     modifier validShares(RevenuePartner[] memory _partners){
-        uint totalShares = 0;            
-        for(uint i = 0; i < _partners.length; i++)
+        uint totalShares = 0;
```

Comment:
zero is default state, no need to assign it (reassignment to same value costs gas.).

Reply:
Done.